Typescript/script.ts

```diff
@@ -0,0 +1,274 @@
+}
+function printHistory(num: string) {
+  document.getElementById("history-value")!.innerText = num;
+}
+function getOutput() {
+  return document.getElementById("output-value")!.innerText;
+}
+function printOutput(num: string) {
+  if (num == "") {
+    y.innerText = num;
+  } else {
+    y.innerText = getFormat(num);
+  }
+}
+function getFormat(num: string) {
+  if (num == "-") {
+    return "";
+  }
+  var n = Number(num);
+  var value = n.toLocaleString("en");
+  return value;
+}
+function reverseNumberFormat(num: string) {
+  return Number(num.replace(/,/g, ""));
+}
+let operator = document.getElementsByClassName("operator");
+for (var i = 0; i < operator.length; i++) {
+  operator[i].addEventListener("click", function (e) {
+    let temp = (<HTMLInputElement>e.target);
+    if (temp.id === "clear") {
+      printHistory("");
+      printOutput("");
+    } else if (temp.id === "backspace") {
+      var output = reverseNumberFormat(getOutput()).toString();
+      if (output) {
+        output = output.substring(0, output.length - 1);
+        printOutput(output);
+      }
+    } else {
+      let output: number | string | boolean = getOutput();
+      let history: any = getHistory();
+      if (output !== "" || output === "" || history !== "") {
+        output = output == "" ? output : reverseNumberFormat(output);
+        history = history + output;
+        if (temp.id === "=") {
+          var result = eval(history);
+          printOutput(result);
+          printHistory("");
+        } else {
+          history = history + temp.id;
+          printHistory(history);
+          printOutput("");
+        }
+      }
+    }
+  });
+}
+let number = document.getElementsByClassName("number");
+for (var i = 0; i < number.length; i++) {
+  number[i].addEventListener("click", function (e) {
+    let temp = (<HTMLInputElement>e.target);
+    let output: number | boolean | string = reverseNumberFormat(getOutput());
+    if (!isNaN(output)) {
+      //if output is a number
+      output = output + temp.id;
+      printOutput(output);
+    }
+  });
+}
+
+let y = document.getElementById("output-value")! as HTMLInputElement;
+function sin() {
+  y.innerText = String(Math.sin(Number(y.innerText)));
+}
+function cos() {
+  y.innerText = String(Math.cos(Number(y.innerText)));
+}
+function tan() {
+  y.innerText = String(Math.tan(Number(y.innerText)));
+}
+function trigo() {
+  let x = document.getElementById("fun")! as HTMLInputElement;
+  switch (x.value) {
+    case "sin":
+      sin();
+      break;
+
+    case "cos":
+      cos();
+      break;
+
+    case "tan":
+      tan();
+      break;
+
+    default:
+      y.innerText = "0";
+      break;
+  }
+}
+function floor() {
+  y.innerText = String(Math.floor(Number(y.innerText)));
+}
+function ceil() {
+  y.innerText = String(Math.ceil(Number(y.innerText)));
+}
+function mathf() {
+  let check = document.getElementById("fun2")! as HTMLInputElement;
+  switch (check.value) {
```

Comment:
In this switch case, create a union type for all possible values of `check.value`